test/fn/watch-project.spec.js

```diff
@@ -0,0 +1,223 @@
+const { expect, assert } = require('chai');
+const sinon = require('sinon');
+const path = require('path');
+const api = require('../api-stub');
+const fs = require('../../src/lib/sync-fs');
+const environment = require('../../src/lib/environment');
+const { watchProject } = require('../../src/fn/watch-project');
+const uploadCustomTranslations = require('../../src/fn/upload-custom-translations').execute;
+const { getTranslationDoc, expectTranslationDocs } = require('./utils');
+
+const testDir = path.join(__dirname, '../data/skeleton');
+const settingsPath = path.join(testDir, 'app_settings', 'base_settings.json');
+const sampleTranslationPath = path.join(testDir, 'translations', 'messages-en.properties');
+const resourceJsonPath = path.join(testDir, 'resources.json');
+const appSettings = fs.readJson(settingsPath);
+
+const mockApi = {
+  updateAppSettings: (content) => {
+    return api.db.put({ _id: 'app_settings', content })
+      .then(() => {
+        return '{"success": true, "updated": true}';
+      });
+  },
+  getAppSettings: () => api.db.get('app_settings')
+};
+
+function editBaseSettings() {
+  return new Promise((resolve) => {
+    const appSettings = fs.readJson(settingsPath);
```

Comment:
The reason for these calls to be wrapped in promises is unclear on first read. 
Could you please add a comment to describe why this is needed?

Reply:
thanks for pointing that out. These functions ares substitutes for the `watch` blocking function and i was under the impression i could only call `await` on Promises https://github.com/medic/cht-conf/blob/b715eea8a62176c82f49c25c6858b05f9ccc4716/src/fn/watch-project.js#L121